gocd/gocd.go

```diff
@@ -0,0 +1,177 @@
+		Name   string `json:"name"`
+		Value  string `json:"value"`
+	} `json:"environment_variables"`
+	Materials []struct {
+		Type       string `json:"type"`
+		Attributes struct {
+			URL             string      `json:"url"`
+			Destination     string      `json:"destination"`
+			Filter          interface{} `json:"filter"`
+			Name            interface{} `json:"name"`
+			AutoUpdate      bool        `json:"auto_update"`
+			Branch          string      `json:"branch"`
+			SubmoduleFolder interface{} `json:"submodule_folder"`
+		} `json:"attributes"`
+	} `json:"materials"`
+	Stages []struct {
+		Name                  string `json:"name"`
+		FetchMaterials        bool   `json:"fetch_materials"`
+		CleanWorkingDirectory bool   `json:"clean_working_directory"`
+		NeverCleanupArtifacts bool   `json:"never_cleanup_artifacts"`
+		Approval              struct {
+			Type          string `json:"type"`
+			Authorization struct {
+				Roles []interface{} `json:"roles"`
+				Users []interface{} `json:"users"`
+			} `json:"authorization"`
+		} `json:"approval"`
+		EnvironmentVariables []interface{} `json:"environment_variables"`
+		Jobs                 []struct {
+			Name                 string        `json:"name"`
+			RunInstanceCount     interface{}   `json:"run_instance_count"`
+			Timeout              interface{}   `json:"timeout"`
+			EnvironmentVariables []interface{} `json:"environment_variables"`
+			Resources            []interface{} `json:"resources"`
+			Tasks                []struct {
+				Type       string `json:"type"`
+				Attributes struct {
+					RunIf            []string    `json:"run_if"`
+					OnCancel         interface{} `json:"on_cancel"`
+					Command          string      `json:"command"`
+					Arguments        []string    `json:"arguments"`
+					WorkingDirectory string      `json:"working_directory"`
+				} `json:"attributes"`
+			} `json:"tasks"`
+			Tabs       []interface{} `json:"tabs"`
+			Artifacts  []interface{} `json:"artifacts"`
+			Properties interface{}   `json:"properties"`
+		} `json:"jobs"`
+	} `json:"stages"`
+	TrackingTool interface{} `json:"tracking_tool"`
+	Timer        interface{} `json:"timer"`
+}
+
+func UnmarshalPipeline(data []byte) Pipeline {
+
+	var pipeline Pipeline
+	err := json.Unmarshal(data, &pipeline)
+	if err != nil {
+		log.Fatalf("ERROR: %v", err)
+	}
+	return pipeline
+}
+
+func UnmarshalPipelineConfig(data []byte) PipelineConfig {
+
+	var pipelineConfig PipelineConfig
+	err := json.Unmarshal(data, &pipelineConfig)
+	if err != nil {
+		log.Fatalf("ERROR: %v", err)
+	}
+	return pipelineConfig
+}
+
+func ReadPipelineJSONFromFile(path string) Pipeline {
+
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return UnmarshalPipeline(data)
+}
+
+func pipelineConfigPOST(gocdULR string, pipelineConfig PipelineConfig) Pipeline {
+
+	// Generated by curl-to-Go: https://mholt.github.io/curl-to-go
+
+	payloadBytes, err := json.Marshal(pipelineConfig)
+	if err != nil {
+		// handle err
```

Comment:
@frank0631 I'd prefer not to commit code that does no error handling at all, could you look into providing some baseline handling of these cases?  It probably would make the most sense for this function to return a Pipeline and an Error, in which case you'd just return (nil, err), but you could also panic with the error message for now if you wish.  See: https://golang.org/doc/effective_go.html#errors